filebeat/input/file/state.go

```diff
@@ -56,83 +56,88 @@ func (s *State) IsEmpty() bool {
 
 // States handles list of FileState
 type States struct {
-	states []State
+	states map[string]State
```

Comment:
@urso Does this change the output format of the registry?

Reply:
No. The registry used to create a copy and print the resulting array. The GetStates() method still returns a copy of `[]State`.

Alternatively I could have introduced some 'indexing' so to have `States` to be like:

```
type States struct {
  idx map[String]int  // map state ID to index into array
  states []State
}
```

Reply:
Seeing the `map[string]State` just reminded me of our first version of the registry where the string was the path and cause lots of issue. Having the unique id here is not a problem I think especially as a unique id will also be able to be created for potentially other state types.